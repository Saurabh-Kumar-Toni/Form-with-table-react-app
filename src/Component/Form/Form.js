import React, { useState } from 'react';

const Form = ({getUsers}) => {

    const [username, setUsername] = useState("");
    const [number, setNumber] = useState("");
    const [gender, setGender] = useState("");

    const handleFormSubmit = (e) => {
        e.preventDefault();
    
        //let items = [...item];
        let obj = {
          username: username,
          number: number, 
          gender: gender
        };
        getUsers(obj);
        setUsername("");
        setNumber("");
        setGender("");
    };

    const handleInputChange = (e) => {

        let name = e.target.name;


        if(name === "number")
        {
          setNumber(e.target.value);
          return;
        }
        else if(name === "username")
        {
          setUsername(e.target.value);
          return;
        }
        else{
          setGender(e.target.value);
        }    
      };

    
    

    return (
      <div id="Form">  
        <form onSubmit={handleFormSubmit}>


          <h2 id="heading">Hello There!!</h2>
          <input id="username" value={username} 
            type="text" name="username" placeholder="Enter your name"
            onChange={handleInputChange} /><br></br>

          
          <input id="number" value={number} 
            type="number" name="number" placeholder="Enter your phone number"
            onChange={handleInputChange} /><br></br>

          <input id="gender" value={gender} 
            type="text" name="gender" placeholder="Enter your Gender"
            onChange={handleInputChange} /><br></br>
          
          <button id="button" type="submit" value="Submit">ADD USER</button>
        </form>
      </div>
    );
  
}

export default Form;