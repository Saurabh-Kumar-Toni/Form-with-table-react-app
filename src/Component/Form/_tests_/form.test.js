import { render, screen } from '@testing-library/react';
import Form from '../Form'

test('check if Form component is rendered properly or not', () => {
 
    const {container} = render(<Form getUsers={jest.fn()}/>)
    expect(container.firstChild).toMatchSnapshot();
  });