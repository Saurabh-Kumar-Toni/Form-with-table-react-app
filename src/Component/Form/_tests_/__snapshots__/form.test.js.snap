// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if Form component is rendered properly or not 1`] = `
<div
  id="Form"
>
  <form>
    <h2>
      Hello There!!
    </h2>
    <input
      id="username"
      name="username"
      placeholder="Enter your name"
      type="text"
      value=""
    />
    <br />
    <input
      id="number"
      name="number"
      placeholder="Enter your phone number"
      type="number"
      value=""
    />
    <br />
    <input
      id="gender"
      name="gender"
      placeholder="Enter your Gender"
      type="text"
      value=""
    />
    <br />
    <button
      id="button"
      type="submit"
      value="Submit"
    >
      ADD USER
    </button>
  </form>
</div>
`;
